retrieval/discovery/zookeeper.go

```diff
@@ -0,0 +1,213 @@
+package discovery
```

Comment:
OK I thought I would do this in a second PR.
How would you like to name the package, as we already import "zk" ?